js/client.js

```diff
@@ -806,12 +813,17 @@
 				var roomTypeLFIndex = roomType.indexOf('\n');
 				if (roomTypeLFIndex >= 0) roomType = roomType.substr(0, roomTypeLFIndex);
 				roomType = toId(roomType);
+
 				if (this.rooms[roomid] || roomid === 'staff') {
 					// staff is always joined in background
 					this.addRoom(roomid, roomType, true);
 				} else {
 					this.joinRoom(roomid, roomType, true);
 				}
+
+				if (roomType === 'chat' && roomid !== 'staff') {
+					this.addAutojoin(roomid, true);
```

Comment:
``addAutojoin`` and ``removeAutojoin`` already do run ``sliceAutojoin`` except when removing a room from the list when its length is less than 6. Would there be a way for the rooms to get out of order in this case, or is the difference in performance between splicing the room out of ``autojoins`` and using ``sliceAutojoins`` not worth the risk?

None of the three methods are actually necessary if no command is ever added.